samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -345,6 +349,22 @@ def _clean_running_containers_and_related_resources(self) -> None:
         cast(WarmLambdaRuntime, self.lambda_runtime).clean_running_containers_and_related_resources()
         cast(RefreshableSamFunctionProvider, self._function_provider).stop_observer()
 
+    def _add_account_id_to_global(self):
+        """
+        Attempts to get the Account ID from the current session
+        If there is no current session, the standard parameter override for
+        AWS::AccountId is used
+        """
+        sts = boto3.client("sts")
```

Comment:
please update to create a client using [`get_boto_client_provider_with_config`](https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/utils/boto_utils.py#L65)? This will ensure the client is created with provided aws profile (which is stored at `self._aws_profile`.

You can see an example of how it's used at https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/delete/delete_context.py#L79-L96

Reply:
Updated